ultradns-rest/src/main/java/denominator/ultradns/service/UltraDNSRestResourceRecordSetApi.java

```diff
@@ -0,0 +1,311 @@
+   */
+  private void add(ResourceRecordSet<?> rrset, int ttlToApply, int typeCode) {
+    LOGGER.debug("Adding record(s) to the zone:" + zoneName + " domain name:" + rrset.name()
+            + " type:" + typeCode);
+    RRSet rrSetNew = new RRSet();
+    rrSetNew.setOwnerName(rrset.name());
+    rrSetNew.setRrtype(rrset.type());
+    rrSetNew.setTtl(ttlToApply);
+    // Extract the rdata from the resource record set passed in as
+    // the parameter to this method.
+    setData(rrSetNew, rrset);
+
+    if (roundRobinPoolApi.isPoolType(rrset.type())) {
+      Profile profile = new Profile();
+      profile.setContext(Constants.RD_POOL_SCHEMA);
+      profile.setOrder(Constants.ROUND_ROBIN);
+      rrSetNew.setProfile(profile);
+    }
+    // Send a POST request.
+    api.createResourceRecord(zoneName, typeCode, rrset.name(), rrSetNew);
+  }
+
+  /**
+   * Updates resource record(s) given rrSetList, resource record set, type, ttlToApply.
+   *
+   * @param rrSetList
+   * @param rrset
+   * @param ttlToApply
+   * @param typeCode
+   * @throws IllegalArgumentException if the zone is not found.
+   */
+  private void replace(RRSetList rrSetList, ResourceRecordSet<?> rrset , int ttlToApply, int typeCode) {
+    LOGGER.debug("Updating resource record(s) of the zone:" + zoneName + " domain name:" + rrset.name()
+            + " type:" + typeCode);
+    // Update the resource record set.
+    List<RRSet> rrSetsExisting = rrSetList.rrSets();
+    RRSet rrSetExisting = rrSetsExisting.remove(0);
+    // Set the TTL value in the resource record set.
+    rrSetExisting.setTtl(ttlToApply);
+    setData(rrSetExisting, rrset);
+    // Send a PUT request.
+    api.updateResourceRecord(zoneName, typeCode, rrset.name(), rrSetExisting);
+  }
+
+  /**
+   * Updates the passed rdatas to the existing RRSet.
+   *
+   * @param rrSet RRSet
+   * @param rrset ResourceRecordSet
+   * @throws IllegalArgumentException if the zone is not found.
+   */
+  private void setData(RRSet rrSet, ResourceRecordSet<?> rrset) {
+    List<String> rdatas = new ArrayList<String>();
+    // Extract the rdata from the resource record set passed in as
+    // the parameter to this method and put them into rdatas.
+    List<Map<String, Object>> toCreate = new ArrayList<Map<String, Object>>(rrset.records());
+    for (Iterator<Map<String, Object>> shouldCreate = toCreate.iterator(); shouldCreate.hasNext();) {
+      String rdata = flatten(shouldCreate.next());
+      rdatas.add(rdata);
+    }
+    rrSet.setRdata(rdatas);
+  }
+
+  /**
+   * Returns RRSetList given owner name and type.
+   *
+   * @param dName
+   * @param type
+   * @throws IllegalArgumentException if the zone is not found.
+   */
+  private RRSetList getRrSetList(String dName, int type) {
+    RRSetList rrSetList;
+    try {
+      rrSetList = api.getResourceRecordsOfDNameByType(zoneName, dName, type);
+    } catch (UltraDNSRestException e) {
+      if (e.code() == UltraDNSRestException.DATA_NOT_FOUND ||
```

Comment:
Throughout the code, you check that if the returned error is not something, you through back the exception. I think its better to create a common method for this and use it everywhere. It can have a parameter for collection of error codes to check & this logic. This way you can kill all those 5-6 lines of repetitive code at numerous places.